### ISSUE TITLE
Returning javascript objects instead of exceptions on error

### ISSUE THREAD

**Issue by ElChache** [Azure/azure-cosmos-js#243]
Hi there

When the library returns an error, like when the given collection doesn't exist, the error itself is not an exception, but a javascript object. It would be easier to retrieve the data and message if it was an exception.

Also, the returned javascript object has a key "body" which is a JSON string (not an object), that needs to be stringified manually, and then the key "message" can be extracted from there.

IMHO, whatever is caught should be a Javascript exception, not an object.

Here is an example that would reproduce the above scenario if the containerName doesn't exist
```
const client = new CosmosClient({ endpoint, auth: { masterKey } });
const database = client.database(databaseId)
const container = database.container(containerName)

container
    .items
    .query(querySpec)
    .toArray()
    .then(function (items) {})
    .catch(function (error) {})
```

**Comment by southpolesteve** [comment]
@ElChache 💯I agree we should be throwing proper Error objects. We are hoping to address this in the next major release of the SDK

**Comment by arichiardi** [comment]
I have run into this as well, it feels unnatural to parse those objects manually in JS land (I have rarely seen throwing anything else than `Error`).

However it would be nice to se that data in the `Error` so it might make sense to include the JSON into a `"data"` property on the `Error` for instance...just as an idea.

**Comment by christopheranderson** [comment]
The v3 error plan is to introduce a few changes:
- an `error.name` property to differentiate errors (rather than status codes, which we were using for some client side errors even if the error didn't come from the backend - which was confusing)
- a property (tbd) which will have diagnostics info on it which is ideally able to be used for some amount of automation (throttling, etc.)
- `Error`/stack trace (but we'll be probably throw it only from the outermost layer of our SDK, and not use exceptions internally).

**Comment by southpolesteve** [comment]
Closed in 3.0.3. We now return `Error` or a subclass of `Error`
